### PR TITLE
(#1586) Guide Card Storybook Update

### LINKED ISSUE #1586: Update storybook structure for nci-guide-card
### `nci-guide-card` storybook structure
1. Default (folder)
    1. Default
    2. Card Group
2. Note: we don't have modifiers, so we don't have a folder for this
3. Variants (folder)
    1. Guide Card with Image and Description (folder)
        2. Single Guide Card with Image and Description without a heading
        3. Title Aligned Card Group
1. Test cases (folder)
    1. Guide Card with Image and Description without Description
    2. Guide with Image and Description without Image
    3. Title Aligned Card Group without Description
    4. Title Aligned Card Group without Image

Other tasks:
* Clean up all of the regressions (redundant breakpoint scenarios) associated with this component
* Create ncids-twig template
* Make sure scenarios still reflect all of the test cases that we have, but we do not need them as examples in the storybook structure

> **ESTIMATE** TBD

### Additional details / screenshot
- [Storybook structure wiki](https://github.com/NCIOCPL/ncids/wiki/Storybook-Structure)

### Related Tickets
* Parent ticket: #1224

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #1586

### CODE REVIEW

**Review by andyvanavery31 — CHANGES_REQUESTED** [review]
Could you please fix the following?

- Update the icons of the "Default", "Test Cases", and "Variants" folders to have the folder icon
- Reorganize the primary subfolders to be Default, Variants, then Test Cases
- Add the "Guide Card with Image and Description" subfolder within the "Variants" folder
- Re-name and reorder the individual items in this to match those listed in the ticket

Could you investigate the following possible redundant regression images and remove ones that you confirm are redundant?
- ncids_nci-guide-card_default_0_document_3_desktop.png and ncids_nci-guide-card_default_0_document_4_widescreen.png
- ncids_nci-guide-card_test-case_with-description-and-without-image_0_document_3_desktop.png and ncids_nci-guide-card_test-case_with-description-and-without-image_0_document_4_widescreen.png
- ncids_nci-guide-card_test-case_with-image-and-without-description_0_document_3_desktop.png and ncids_nci-guide-card_test-case_with-image-and-without-description_0_document_4_widescreen.png
- ncids_nci-guide-card_with-image-and-description_default_0_document_3_desktop.png and ncids_nci-guide-card_with-image-and-description_default_0_document_4_widescreen.png

**Comment by andyvanavery31** [comment]
Last three minor changes: 

1. Can you add a space between "GuideCard with Image and Description" folder that is inside the "Variants" folder?
2. Can you remove the space between "With Out" that is in the story title "Single Card With Image And Description With Out Heading" which is inside the "GuideCard with Image and Description" folder inside "Variants"?
3. Can you make all instances of "With", "Without", and "And" all lowercase? (i.e. change "Title Aligned Card Group Without Description" to "Title Aligned Card Group without Description")

**Review by andyvanavery31 — APPROVED** [review]
Looks good to me!

**Review by olitharp-nci — CHANGES_REQUESTED** [review]
This has the same notes from card

File: testing/ncids-css-testing/.backstop/reference/ncids_nci-guide-card_default_0_document_1_tablet.png
Comment (olitharp-nci) [inline testing/ncids-css-testing/.backstop/reference/ncids_nci-guide-card_default_0_document_1_tablet.png:1]: the paddings changing on all of these are concerning -- are these expected changes?

File: testing/ncids-css-testing/.backstop/reference/ncids_nci-guide-card_variants_with-image-and-description_title-aligned_card-group_0_document_3_desktop.png
Comment (olitharp-nci) [inline testing/ncids-css-testing/.backstop/reference/ncids_nci-guide-card_variants_with-image-and-description_title-aligned_card-group_0_document_3_desktop.png:1]: fix these?

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: testing/ncids-css-testing/stories/components/nci-guide-card/nci-guide-card.testcases.description.without.image.stories.jsx
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/components/nci-guide-card/nci-guide-card.testcases.description.without.image.stories.jsx:1]: * nci-guide-card.default.jsx
  * `title: 'components/Guide Card/Default',`
  * `export const Default ...`
  * `export const CardGroup ...`
* nci-guide-card.variants.jsx
  * `title: 'components/Guide Card/Variants/Guide Card With Image and Description',`
  * `export const SingleGuideCardWithImageAndDescriptionWithoutHeading ...`
  * `export const TitleAlignedCardGroup ...`
* nci-guide-card.test-cases.jsx
  * `title: 'components/Guide Card/Test Cases',`
  * `export const GuideCardWithImageAndDescriptionWithoutDescription ...`
  * `export const Guide with Image and DescriptionWithoutImage ...`
  * `export const TitleAlignedCardGroupWithoutDescription ...`
  * `export const TitleAlignedCardGroupWithoutImage ...`

(who named these variants)

**Review by olitharp-nci — CHANGES_REQUESTED** [review]
nci-guide-card.testcases.title.aligned.group.without.description.stories.jsx

these file names are outta pocket try the following

File: testing/ncids-css-testing/.backstop/reference/ncids_nci-guide-card_default_0_document_2_tablet-lg.png
Comment (olitharp-nci) [inline testing/ncids-css-testing/.backstop/reference/ncids_nci-guide-card_default_0_document_2_tablet-lg.png:1]: The spacings are changing so much on these we either need design to approve or something in the template needs to be adjusted so there are less regressions @blilianyu

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by olitharp-nci — CHANGES_REQUESTED** [review]
@andyvanavery31 please review the regressions.

**Review by andyvanavery31 — APPROVED** [review]
Looks good to me!

**Comment by andyvanavery31** [comment]
Please update the padding/spacing of the guide cards to match Prod @karlikpj

**Review by andyvanavery31 — APPROVED** [review]
The stories and regression images match prod. Looks good to me!

**Comment by olitharp-nci** [comment]
Changes for quick merge: https://github.com/NCIOCPL/ncids/compare/6ba5e783cb334fbf4130cf68519fdca7860d345b..d2ee444cb8317335f477876e38828f2457e76df2
* This compare also includes the rebase, so it hard to tell the changes, which include basic formatting again. It appears to be an ongoing issue for twig files. I will add a ticket to look into missing linting in twig files.

**Review by olitharp-nci — APPROVED** [review]

**Pull request merged**